### PR TITLE
DT-2825: ⬆️  Switch to java 17

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

**Review by Mjwillis — APPROVED** [review]
LGTM

File: build.gradle [hunk 1]
```diff
@@ -22,38 +22,38 @@ dependencies {
...
     runtimeOnly 'com.h2database:h2:1.4.200'
     testRuntimeOnly 'com.h2database:h2:1.4.200'
-    testAnnotationProcessor 'org.projectlombok:lombok:1.18.20'
-    testCompileOnly 'org.projectlombok:lombok:1.18.20'
+    testAnnotationProcessor 'org.projectlombok:lombok:1.18.22'
+    testCompileOnly 'org.projectlombok:lombok:1.18.22'
 
     testImplementation 'org.codehaus.groovy.modules.http-builder:http-builder:0.7.1'
-    testImplementation ('io.rest-assured:rest-assured:3.3.0'){
+    testImplementation ('io.rest-assured:rest-assured:4.4.0'){
```
Comment (andymarke) [inline build.gradle:52]: if the tests work with this that fantastic, for some reason I though the tests would break on v4

**Review by andymarke — APPROVED** [review]
LGTM

File: build.gradle (hunk 1 above)
Comment (petergphillips) [reply build.gradle:52]: I'm sure I tried it previously and they didn't work, but they all appear to have passed :grin:

**Pull request merged**